listen_test.go

```diff
@@ -115,6 +115,118 @@ func Test_Listen_Graceful_Shutdown(t *testing.T) {
 	mu.Unlock()
 }
 
+// go test -run Test_Listen_Graceful_Shutdown_Timeout
+func Test_Listen_Graceful_Shutdown_Timeout(t *testing.T) {
+	var mu sync.Mutex
+	var shutdownSuccess bool
+	var shutdownTimeoutError error
+
+	app := New()
+
+	app.Get("/", func(c Ctx) error {
+		return c.SendString(c.Hostname())
+	})
+
+	ln := fasthttputil.NewInmemoryListener()
+	errs := make(chan error)
+
+	go func() {
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+		defer cancel()
+
+		errs <- app.Listener(ln, ListenConfig{
+			DisableStartupMessage:   true,
+			GracefulContext:         ctx,
+			GracefulShutdownTimeout: 500 * time.Millisecond,
+			OnShutdownSuccess: func() {
+				mu.Lock()
+				shutdownSuccess = true
+				mu.Unlock()
+			},
+			OnShutdownError: func(err error) {
+				mu.Lock()
+				shutdownTimeoutError = err
+				mu.Unlock()
+			},
+		})
+	}()
+
+	// Server readiness check
+	for i := 0; i < 10; i++ {
+		conn, err := ln.Dial()
+		// To test a graceful shutdown timeout, do not close the connection.
+		if err == nil {
+			_ = conn
+			break
+		}
+		// Wait a bit before retrying
+		time.Sleep(100 * time.Millisecond)
+		if i == 9 {
+			t.Fatalf("Server did not become ready in time: %v", err)
+		}
+	}
+
+	testCases := []struct {
+		ExpectedErr             error
+		ExpectedShutdownError   error
+		ExpectedBody            string
+		Time                    time.Duration
+		ExpectedStatusCode      int
+		ExpectedShutdownSuccess bool
+	}{
+		{
+			Time:                    100 * time.Millisecond,
+			ExpectedBody:            "example.com",
+			ExpectedStatusCode:      StatusOK,
+			ExpectedErr:             nil,
+			ExpectedShutdownError:   nil,
+			ExpectedShutdownSuccess: false,
+		},
```

Comment:
Can you add a test-case for shutdown after 1 second

Reply:
And i think all tests should start and stop the fiber server independently. They should not affect each other since we use sleep